pkg/pipelines/git/repository_test.go

```diff
@@ -0,0 +1,257 @@
+package git
+
+import (
+	"fmt"
+	"net/url"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/h2non/gock"
+)
+
+var mockHeaders = map[string]string{
+	"X-GitHub-Request-Id":   "DD0E:6011:12F21A8:1926790:5A2064E2",
+	"X-RateLimit-Limit":     "60",
+	"X-RateLimit-Remaining": "59",
+	"X-RateLimit-Reset":     "1512076018",
+}
+
+func TestWebhookWithFakeClient(t *testing.T) {
```

Comment:
This should be broken out into separate tests, this is multiple tests in one.

Reply:
You can't unless you construc Fake client with new.DefaultClient()  If you do that, the tests will create/delete webhooks directly with Fake Client.   You will end up testing the Fake Cleint only.    We want to create a scm client likes this test code.
repo, err := NewRepository("https://fake.com/foo/bar.git", "token")

Reply:
Fake client doesn't need at all.

Reply:
You either call new.DefaultClient() or call scm client by factory.   You need DefaultClient() if you want to setup your Fake Data.    You need to set up your Fake Data if you want break down the tests into create test, delete tests ,etc.    You can't test delete() if there is nothing in the Fake Data.